consensus/istanbul/config.go

```diff
@@ -16,29 +16,99 @@
 
 package istanbul
 
-import "math/big"
+import (
+	"math/big"
 
-type ProposerPolicy uint64
+	"github.com/naoina/toml"
+)
+
+type ProposerPolicyId uint64
 
 const (
-	RoundRobin ProposerPolicy = iota
+	RoundRobin ProposerPolicyId = iota
 	Sticky
 )
 
+// ProposerPolicy represents the Validator Proposer Policy
+type ProposerPolicy struct {
+	Id       ProposerPolicyId    // Could be RoundRobin or Sticky
+	Registry []ValidatorSet      // Holds the ValidatorSet for a given block height
+	By       validatorSortByFunc // func that defines how the ValidatorSet should be sorted
+}
+
+// NewRoundRobinProposerPolicy returns a RoundRobin ProposerPolicy with ValidatorSortByStringFunc as default sort function
+func NewRoundRobinProposerPolicy() *ProposerPolicy {
+	return NewProposerPolicy(RoundRobin)
+}
+
+// NewStickyProposerPolicy return a Sticky ProposerPolicy with ValidatorSortByStringFunc as default sort function
+func NewStickyProposerPolicy() *ProposerPolicy {
+	return NewProposerPolicy(Sticky)
+}
+
+func NewProposerPolicy(id ProposerPolicyId) *ProposerPolicy {
+	return &ProposerPolicy{Id: id, By: ValidatorSortByStringFunc}
+}
+
+type proposerPolicyToml struct {
```

Comment:
may be able to achieve the same with `toml:"-"` struct tags in `ProposerPolicy`

Reply:
The intermediary struct `proposerPolicyToml` is still required, otherwise the Unmarshall goes into an infinite loop of calling itself